stardog/admin.py

```diff
@@ -752,6 +756,161 @@ def cluster_info(self):
+          *names: (str): Names of the cached graphs or queries
+        Returns:
+          list[str]: List of statuses
+        """
+        return self.client.post('/admin/cache/status', json=names).json()
+
+    def cached_queries(self):
+        """Retrieves all cached queries.
+
+        Returns:
+          list[Cache]: A list of Cache objects
+        """
+        r = self.client.get('/admin/cache/queries')
+        cache_names = [cache_name['name'] for cache_name in r.json()]
+        return list(map(lambda name: Cache(name, self.client), cache_names))
+
+    def cached_graphs(self):
+        """Retrieves all cached graphs.
+
+        Returns:
+          list[Cache]: A list of Cache objects
+        """
+        r = self.client.get('/admin/cache/graphs')
+        cache_names = [cache_name['name'] for cache_name in r.json()]
+        return list(map(lambda name: Cache(name, self.client), cache_names))
+
+
+    def new_cached_query(self, name, target, query, database, refresh_script=None, register_only=False):
+        """Creates a new cached query.
+
+        Args:
+          name (str): The name (URI) of the cached query
+          target (str): The name (URI) of the cache target
+          query (str): The query to cache
+          database (str): The name of the database
+          refresh_script (str, optional): A SPARQL insert query to run
+            when refreshing the cache
+          register_only (bool): Default: false. If true, register a
+            cached dataset without loading data from the source graph
+            or query into the cache target's databases
+
+        Returns:
+          Cache: The new Cache
+        """
+
+        params = {
+            'name': name,
+            'target': target,
+            'database': database,
+            'query': query,
+
+        }
+
+        if refresh_script:
+            params['refreshScript'] = refresh_script
+
+        if register_only:
+            params['registerOnly'] = True
+
+        self.client.post('/admin/cache', json=params)
+        return Cache(name, self.client)
+
+    def new_cached_graph(self, name, target, graph, database=None, refresh_script=None, register_only=False):
+        """Creates a new cached graph.
+
+        Args:
+          name (str): The name (URI) of the cached query
+          target (str): The name (URI) of the cache target
+          graph (str): The name of the graph to cache
+          database (str): The name of the database. Optional for virtual graphs, mandatory for named graphs.
+          refresh_script (str): An optional SPARQL Insert query to run when refreshing the cache.
+          register_only (bool): An optional value that if true, register a cached dataset without loading data from the source graph or query into the cache target's databases.
+
+
+        Returns:
+          Cache: The new Cache"""
+
+        # database is optional for VG, mandatory for named graphs
+
+        if type(register_only) is not bool:
```

Comment:
why is this type specifically checked?

Reply:
same as before. it can be removed, I am just thinking in the case that someone can pass `'False'` as string and not as bool.

Reply:
I think that is just the world python lives in

Reply:
removed